apps/shipments/management/commands/replicate_shipments.py

```diff
@@ -0,0 +1,97 @@
+from apps.shipments.models import Shipment
+
+logger = logging.getLogger('transmission')
+logger.setLevel(settings.LOG_LEVEL)
+
+
+class Command(BaseCommand):
+    help = 'Replicate shipment related transactions to the sidechain.'
+    rpc_client = None
+    successfull_shipments = []
+    unsuccessfull_shipments = []
+    async_job = None
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--shipment_id',
+            type=str,
+            default=None,
+            help='Replicate specific shipment instead of all of them.',
+        )
+
+    def _replicate_shipment(self, shipment):
+        self.successfull_shipments.append(shipment.id)
+        for async_job in AsyncJob.objects.filter(shipment=shipment):
+            try:
+                eth_action = EthAction.objects.filter(async_job=async_job).first()
+                if eth_action:
+                    logger.info(f'Deleting Eth Action: {eth_action.transaction_hash}')
+                    eth_action.delete()
+                AsyncTask(async_job.id).rerun()
+                self.async_job = async_job
+
+            # pylint:disable=broad-except
+            except Exception as exc:
+                if shipment.id in self.successfull_shipments:
+                    self.successfull_shipments.remove(shipment.id)
+                    self.unsuccessfull_shipments.append(shipment.id)
+                logger.error(exc)
+
+    def _unsubscribe(self):
+        self.rpc_client = EventRPCClient()
+        try:
+            self.rpc_client.unsubscribe("LOAD", settings.LOAD_VERSION)
+        except RPCError as exc:
+            if 'Error: EventSubscription not found' not in exc.detail:
+                raise exc
+
+    def _resubscribe(self):
+        if not self.async_job:
+            logger.warning('Unable to find last blockheight to subscribe to. '
+                           'Need to manually subscribe to latest blockheight')
+            return
+
+        message = Message.objects.filter(async_job=self.async_job).last()
+        if "blockNumber" not in message.body:
+            time.sleep(15)
+            message = Message.objects.filter(async_job=self.async_job).last()
+        if "blockNumber" in message.body:
+            self.rpc_client.subscribe("LOAD", settings.LOAD_VERSION, last_block=message.body["blockNumber"])
+        else:
+            logger.warning('Unable to find last blockheight to subscribe to. '
+                           'Need to manually subscribe to latest blockheight')
+
+    def handle(self, *args, **options):
+        self._unsubscribe()
+        if options['shipment_id']:
+            shipment = Shipment.objects.filter(id=options['shipment_id']).first()
+            if not shipment:
+                logger.error(f'Invalid shipment id: {options["shipment_id"]} supplied')
+                sys.exit()
+
+            self._replicate_shipment(shipment)
+
+        else:
+            shipments = Shipment.objects.all()
```

Comment:
Is there anything we can filter down to only fetch unmigrated shipments? In case this fails part of the way thru and we need to kick it off again

Reply:
One way to do this is to inspect the `Message.body` field for the associated `AsyncJobs` being processed.  If the `cumulativeGasUsed` for a transaction is `0` then this transaction ran on the sidechain since transactions do not incur a gas cost.

Reply:
Would it need to check every message.body or would just checking the last one, like how this function obtains the `blockNumber` to resubscribe with